File: app/src/main/java/com/neubofy/veto/utils/SecureSettings.java
package com.neubofy.veto.utils;

import android.content.Context;
import android.provider.Settings;

public class SecureSettings {

    private static final String TAG = SecureSettings.class.getSimpleName();

    public static void turnGPS(Context context, boolean enable) {
        int value;
        if (enable) {
            value = android.provider.Settings.Secure.LOCATION_MODE_HIGH_ACCURACY;
        } else {
            value = android.provider.Settings.Secure.LOCATION_MODE_OFF;
        }
        Settings.Secure.putString(context.getContentResolver(), android.provider.Settings.Secure.LOCATION_MODE, Integer.valueOf(value).toString());
        VetoLogKt.log(context).d(TAG, "Turned GPS on/off using SecureSettings: " + enable);
    }

    public static boolean setBluetooth(Context context, boolean enable) {
        try {
            int value = enable ? 1 : 0;
            Settings.Global.putInt(context.getContentResolver(), "bluetooth_on", value);
            VetoLogKt.log(context).d(TAG, "Turned Bluetooth on/off using SecureSettings: " + enable);
            return true;
        } catch (Exception e) {
            VetoLogKt.log(context).e(TAG, "Failed to toggle bluetooth via SecureSettings: " + e.getMessage());
            return false;
        }
    }
}

File: app/src/main/java/com/neubofy/veto/utils/SecureSettings.kt
package com.neubofy.veto.utils

import android.content.Context
import android.provider.Settings
import com.neubofy.veto.utils.log

object SecureSettings {
    private val TAG = SecureSettings::class.java.simpleName

    @JvmStatic
    fun turnGPS(context: Context, enable: Boolean) {
        @Suppress("DEPRECATION")
        val value = if (enable) {
            Settings.Secure.LOCATION_MODE_HIGH_ACCURACY
        } else {
            Settings.Secure.LOCATION_MODE_OFF
        }
        @Suppress("DEPRECATION")
        Settings.Secure.putString(
            context.contentResolver,
            Settings.Secure.LOCATION_MODE,
            value.toString()
        )
        context.log().d(TAG, "Turned GPS on/off using SecureSettings: $enable")
    }

}
